tests/pipeline/test_quarters_estimates.py

```diff
@@ -0,0 +1,747 @@
+from abc import abstractmethod
+import blaze as bz
+import itertools
+from nose_parameterized import parameterized
+import numpy as np
+import pandas as pd
+from toolz import merge
+
+from zipline.pipeline import SimplePipelineEngine, Pipeline, CustomFactor
+from zipline.pipeline.common import (
+    EVENT_DATE_FIELD_NAME,
+    FISCAL_QUARTER_FIELD_NAME,
+    FISCAL_YEAR_FIELD_NAME,
+    SID_FIELD_NAME,
+    TS_FIELD_NAME,
+)
+from zipline.pipeline.data import DataSet, Column
+from zipline.pipeline.loaders.blaze.estimates import (
+    BlazeNextEstimatesLoader,
+    BlazePreviousEstimatesLoader
+)
+from zipline.pipeline.loaders.quarter_estimates import (
+    NextQuartersEstimatesLoader,
+    PreviousQuartersEstimatesLoader,
+    split_normalized_quarters, normalize_quarters)
+from zipline.testing import ZiplineTestCase
+from zipline.testing.fixtures import WithAssetFinder, WithTradingSessions
+from zipline.testing.predicates import assert_equal
+from zipline.utils.numpy_utils import datetime64ns_dtype, float64_dtype
+
+
+class Estimates(DataSet):
+    event_date = Column(dtype=datetime64ns_dtype)
+    fiscal_quarter = Column(dtype=float64_dtype)
+    fiscal_year = Column(dtype=float64_dtype)
+    estimate = Column(dtype=float64_dtype)
+
+
+def QuartersEstimates(num_qtr):
+    class QtrEstimates(Estimates):
+        num_quarters = num_qtr
+        name = Estimates
+    return QtrEstimates
+
+
+# 0Q1: 2015-01-05.Q1.e1.2015-01-06, 2015-01-10.Q1.e1.2015-01-11,
+# 0Q2: 2015-01-15.Q2.e1.2015-01-16, 2015-01-20.Q2.e1.2015-01-21,
+# 0Q4: 2015-02-05.Q4.e1.2015-02-06, 2015-02-10.Q4.e1.2015-02-11,
+# Skip Q3 to make sure we handle skipped quarter data correctly.
+estimates_timeline = pd.DataFrame({
+    TS_FIELD_NAME: [pd.Timestamp('2015-01-05'), pd.Timestamp('2015-01-07'),
+                    pd.Timestamp('2015-01-05'), pd.Timestamp('2015-01-17'),
+                    pd.Timestamp('2015-01-05'), pd.Timestamp('2015-01-17'),
+                    pd.Timestamp('2015-01-22'), pd.Timestamp('2015-02-02')],
+    EVENT_DATE_FIELD_NAME:
+        [pd.Timestamp('2015-01-10'), pd.Timestamp('2015-01-10'),
+         pd.Timestamp('2015-01-20'), pd.Timestamp('2015-01-20'),
+         pd.Timestamp('2015-02-10'), pd.Timestamp('2015-02-10'),
+         pd.Timestamp('2015-02-10'), pd.Timestamp('2015-02-10')],
+    'estimate': [1.]*2 + [2.] * 2 + [4.] * 4,
+    FISCAL_QUARTER_FIELD_NAME: [1]*2 + [2] * 2 + [4] * 4,
+    FISCAL_YEAR_FIELD_NAME: [2015]*8,
+    SID_FIELD_NAME: [0]*8
```

Comment:
My thinking was that this way it's easier to see how the data corresponds to the cases and that it's more concise.